assignment1/string_permutation.py

```diff
@@ -0,0 +1,18 @@
+import string 
+
+def string_permutation(str1, str2):
+    if (str.isalpha(str1) or str1 == "") and (
+            str.isalpha(str2) or str2 == ""):
+        str1 = str1.lower()
+        str2 = str2.lower()
+        inputs = {x: 0 for x in string.ascii_lowercase} 
+        for c in str1: 
+            inputs[c] += 1
+        for c in str2: 
+            inputs[c] -= 1
+        for c in inputs.keys():
```

Comment:
Sometimes list comprehensions can be cryptic but it might be straightforward enough to use one here:

```
return all(c == 0 for c in inputs.values())
```